src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -47,6 +47,61 @@ function CompleteBipartiteGraph(n1::T, n2::T) where {T <: Integer}
     return SimpleGraph(ne, fadjlist)
 end
 
+"""
+    CompleteMultipartiteGraph(partitions)
+
+Create an undirected [complete bipartite graph](https://en.wikipedia.org/wiki/Complete_bipartite_graph)
+with `sum(partitions)` vertices and `prod(partitions)` edges. A partition with `0` vertices is skipped.
+"""
+function CompleteMultipartiteGraph(partitions::AbstractVector{T}) where {T <: Integer}
+    any(x -> x < 0, partitions) && return SimpleGraph{T}(0)
+    length(partitions) == 1 && return SimpleGraph{T}(partitions[1])
+    length(partitions) == 2 && return CompleteBipartiteGraph(partitions[1], partitions[2])
+
+    Tw = widen(T)
+    nw = sum(partitions)
+    n = T(nw)  # checks if T is large enough for sum(partitions)
```

Comment:
Why is this type coercion necessary? Can't we just do `nw = sum(partitions)` and then just get the type of `nw` if we need it?

That is, let the types widen naturally:
```
julia> a = Vector{Int8}([254, 254, 254, 254, 254])^C

julia> a = Vector{Int8}([127,127,127,127,127]);

julia> b = sum(a)
635

julia> typeof(b)
Int64
```

Reply:
Sorry, that was a remnant from when I was copying the style of CompleteBipartite. This raises the question of why CompleteBipartite graph shouldn't be promoted to the next largest integer type if n1+n2 causes overflow?

Reply:
I would check out some of the operators.jl functions. IIRC, in some cases we say explicitly that overflow will error. In others you could consider using `squash()`.